revision.c

```diff
@@ -3342,6 +3377,32 @@ static void expand_topo_walk(struct rev_info *revs, struct commit *commit)
 	}
 }
 
+static void prepare_to_use_bloom_filter(struct rev_info *revs)
+{
+	if (!revs->commits)
+	    return;
+
+	parse_commit(revs->commits->item);
+
+	if (!revs->repo->objects->commit_graph)
+		return;
+
+	revs->bloom_filter_settings = revs->repo->objects->commit_graph->settings;
+	if (!revs->bloom_filter_settings)
+		return;
+
```

Comment:
Here:

```
if (revs->bloom_filter_settings->hash_version != 1) {
    warning(_("unrecognized changed paths data version: %d"),
             revs->bloom_filter_settings->hash_version);
    return;
}
```

Reply:
(Or better, do this when you are reading the settings from memory. check the first 4 bytes match `1`, then don't load settings at that point. Perhaps don't even warn.)

Reply:
I like doing this during the initial load when parsing the graph. I will do this in a fix up commit and apply it to the appropriate older commit. Not taking this change as part of this commit. @derrickstolee

Reply:
Also, thank you for thinking of this.